### PR TITLE
🌐 Add French translation for `docs/fr/docs/index.md`

### PR DESCRIPTION
**Pull request by cfraboulet**
Hello 👋

This is a PR to translate `docs/index.md` to french 🇫🇷
As a reminder, I took the previous PRs (https://github.com/tiangolo/fastapi/pull/1973 and https://github.com/tiangolo/fastapi/pull/4203) and combine them into this new PR in order to close the previous ones.

According to the discussion: https://github.com/tiangolo/fastapi/discussions/9185.

Thanks for the reviews 🙏

### CODE REVIEW

**Comment by frabc** [comment]
Hi @tiangolo 😊 If this PR receives the two necessary approvals, the PRs https://github.com/tiangolo/fastapi/pull/1973 and https://github.com/tiangolo/fastapi/pull/4203 can be closed.

Thanks 🙏🏻

**Review by axel584 — APPROVED** [review]

File: docs/fr/docs/index.md [hunk 1]
```diff
@@ -1,48 +1,46 @@
...
-**Documentation**: <a href="https://fastapi.tiangolo.com" target="_blank">https://fastapi.tiangolo.com</a>
+**Documentation** : <a href="https://fastapi.tiangolo.com" target="_blank">https://fastapi.tiangolo.com</a>
 
-**Source Code**: <a href="https://github.com/tiangolo/fastapi" target="_blank">https://github.com/tiangolo/fastapi</a>
+**Code Source** : <a href="https://github.com/tiangolo/fastapi" target="_blank">https://github.com/tiangolo/fastapi</a>
 
 ---
 
-FastAPI is a modern, fast (high-performance), web framework for building APIs with Python 3.6+ based on standard Python type hints.
+FastAPI est un framework web moderne et rapide (haute performance) pour la construction d'API avec Python 3.7+, basé sur les annotations de type standard de Python.
```
Comment (rjNemo) [inline docs/fr/docs/index.md:30]: I would rather use `creation` instead of `construction`

File: docs/fr/docs/index.md [hunk 2]
```diff
@@ -1,48 +1,46 @@
...
 
-The key features are:
+Les principales fonctionnalités sont :
 
-* **Fast**: Very high performance, on par with **NodeJS** and **Go** (thanks to Starlette and Pydantic). [One of the fastest Python frameworks available](#performance).
+* **Rapidité** : De très hautes performances, à égalité avec **NodeJS** et **Go** (grâce à Starlette et Pydantic). [L'un des frameworks Python les plus rapides](#performance).
```
Comment (rjNemo) [inline docs/fr/docs/index.md:34]: what about `au niveau de` for `à égalité`

**Comment by frabc** [comment]
Hi @rjNemo 😊 I applied your suggestions! Thanks for your returns.

**Review by rjNemo — APPROVED** [review]

File: docs/fr/docs/index.md [hunk 3]
```diff
@@ -189,15 +193,15 @@ async def read_item(item_id: int, q: Union[str, None] = None):
     return {"item_id": item_id, "q": q}
 ```
 
-**Note**:
+**Note**
 
-If you don't know, check the _"In a hurry?"_ section about <a href="https://fastapi.tiangolo.com/async/#in-a-hurry" target="_blank">`async` and `await` in the docs</a>.
+Si vous n'y êtes pas familiers, consultez la section _"Vous êtes pressés ?"_ à propos de <a href="https://fastapi.tiangolo.com/fr/async/#vous-etes-presses" target="_blank">`async` et `await` dans la documentation</a>.
```
Comment (Viicos) [inline docs/fr/docs/index.md:198]: `Si vous n'êtes pas familier` ?

File: docs/fr/docs/index.md [hunk 4]
```diff
@@ -214,56 +218,56 @@ INFO:     Application startup complete.
...
 </details>
 
-### Check it
+### Vérifiez
 
-Open your browser at <a href="http://127.0.0.1:8000/items/5?q=somequery" class="external-link" target="_blank">http://127.0.0.1:8000/items/5?q=somequery</a>.
+Ouvrez votre navigateur à l'adresse <a href="http://127.0.0.1:8000/items/5?q=somequery" class="external-link" target="_blank">http://127.0.0.1:8000/items/5?q=somequery</a>.
 
-You will see the JSON response as:
+Vous obtenez cette réponse <abbr title="JavaScript Object Notation">JSON</abbr> :
```
Comment (Viicos) [inline docs/fr/docs/index.md:235]: `Vous obtenez alors`

**Comment by frabc** [comment]
Hi @Viicos 😊 I applied your suggestions! Thanks for your returns.

**Comment by tiangolo** [comment]
Awesome, thank you @frabc! :bow: 

And thanks for the reviews @Viicos, @axel584, @rjNemo! :coffee:

**Pull request merged**